### DISCUSSION TITLE
Question about expanding streams (scaling).

### DISCUSSION THREAD

**Discussion by epigraphe** [roboflow/supervision#393] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

Now I am using your example, processed for launching on streamlit. Everything works as expected, the camera sees people and counts them.
Now I need to scale the system. Roughly speaking, on the home page of Streamlit I will have 6 cameras, each of which should detect people and display their current number.
Is this possible within one server? If I simply copy a working recognition block 6 times within one script, will everything work together?
I understand that this will take a large amount of resources, we are ready to purchase video cards and other power for this, but the question here will be, will this work in principle?

**Comment by SkalskiP** [comment]
Hi, @epigraphe 👋🏻 Do you run inference locally or over the API?

**Comment by epigraphe** [comment]
Everything works on one machine and with one script. I don’t want to flood and I won’t copy it again))), I already posted an example of the script earlier, here.
https://github.com/roboflow/supervision/issues/390#issue-1923319481

Now it works on a CPU with an 8n model, we plan to transfer it to a GPU

**Comment by onuralpszr** [comment]
@epigraphe If you want to run for 6 camera what you can do is If you wanna do via "ultralytics" you 
can do "multhread-tracking" example explain [here](https://docs.ultralytics.com/modes/track/#multithreaded-tracking) or you can setup a inference server and call via that as well.

**Reply by SkalskiP** [reply]
You can also consider deploying your model via [inference](https://github.com/roboflow/inference) and interacting over the API. Running on 6 cameras locally might be problematic. It's a lot of frames (6 * 30 = 180) every second.

**Comment by epigraphe** [comment]
Thank you. Will there be any problems from supervision?
For example, will I be able to easily get the number of people in a frame and use this data in the future?

**Reply by SkalskiP** [reply]
I don't expect any problems. Supervision should work with no problems. If you will face any problems we are here to help.
